### ISSUE TITLE
[CONTENT] Lazy minting

### ISSUE THREAD

**Issue by jessicaschilling** [protocol/nft-website#97]
This issue is for drafting content for the following:
- Title: Lazy minting
- Address: `/how-to/lazy-minting/`

**Comment by jessicaschilling** [comment]
@yusefnapora -- here's your replacement quest for the week 😊

**Comment by yusefnapora** [comment]
I'm still in research mode for this, but I wanted to toss in a quick status update.

I've been following along with a great [OpenZeppelin workshop](https://blog.openzeppelin.com/workshop-recap-building-an-nft-merkle-drop/) on lazy minting techniques, and I think I get the basic premise now :)

The key to the lazy flow seems to be that the cost to mint is payed by the purchaser at the time of purchase, without the minter needing to send a transaction and pay gas first. This works by having the buyer "redeem" a signature that was created by an Ethereum private key that belongs to an authorized "minter" account defined in the contract. 

Basically, instead of sending a "mint token" transaction to create a new NFT, the minter account signs a hash of the token id and gives the signature to the buyer. The buyer then puts the signature into the contract's `redeem` function which validates that it was produced by someone authorized to mint, then mints the NFT and assigns it to the buyer.

Anyway, I've been playing with this and trying to get something working locally. will update here with progress

**Comment by jessicaschilling** [comment]
Thanks for the update!

**Comment by jessicaschilling** [comment]
Per sync discussion — @yusefnapora will scaffold this for @johndmulhausen to build out later this week. Thanks both!

**Comment by yusefnapora** [comment]
I just got this to work today, and I've started drafting a bit of the text. So far I've been working in a completely separate sandbox repo, but I'll be moving the code over to the nft-school-examples repo now that things seem to be working. I'll also open a draft PR here for the tutorial. I'm about out of gas though, so that will probably happen on Monday morning.

The WIP repo is here if anyone wants to check it out: https://github.com/yusefnapora/lazy-minting

There's no UI to speak of, just some unit tests in `lazy-mint-example/test/lazy-test.js`. If you want to sanity check that it works on your machine, you could `cd lazy-mint-example` and give it a quick `npm install`, and then run `npx hardhat test`.

@jessicaschilling / @johndmulhausen If you have any feedback on the README, lmk. It's not finished & needs some details on the signatures and so on, but we should be able to use most of it for the article, I think.
